catalog/to-consul/syncer_test.go

```diff
@@ -155,6 +155,45 @@ func TestConsulSyncer_reapService(t *testing.T) {
 	}
 }
 
+// Test that the syncer doesn't reap any services until the initial sync has
+// been performed.
+func TestConsulSyncer_noReapingUntilInitialSync(t *testing.T) {
+	t.Parallel()
+
+	a, err := testutil.NewTestServerT(t)
+	require.NoError(t, err)
+	defer a.Stop()
+	client, err := api.NewClient(&api.Config{
+		Address: a.HTTPAddr,
+	})
+	require.NoError(t, err)
+	s, closer := testConsulSyncer(client)
+	defer closer()
+
+	// Create a service directly in Consul. Since it was created directly we
+	// expect it to be deleted but not until the initial sync is performed.
```

Comment:
```suggestion
	// Create a service directly in Consul. Since it was created on the 
	// synthetic sync node and has the sync-associated tag, we expect
	// it to be deleted but not until the initial sync is performed.
```

A service needs the sync tag and to be registered on the sync node or the sync process would ignore it.